server_installation/topics/operating-mode/crossdc.adoc

```diff
@@ -0,0 +1,736 @@
+    <remote-store cache="loginFailures" remote-servers="remote-cache" passivation="false" fetch-state="false" purge="false" preload="false" shared="true">
+        <property name="rawValues">true</property>
+        <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
+    </remote-store>
+</distributed-cache>
+
+<distributed-cache name="actionTokens" mode="SYNC" owners="2">
+    <eviction max-entries="-1" strategy="NONE"/>
+    <expiration max-idle="-1" interval="300000"/>
+    <remote-store cache="actionTokens" remote-servers="remote-cache" passivation="false" fetch-state="false" purge="false" preload="true" shared="true">
+        <property name="rawValues">true</property>
+        <property name="marshaller">org.keycloak.cluster.infinispan.KeycloakHotRodMarshallerFactory</property>
+    </remote-store>
+</distributed-cache>
+```
++
+
+.. Add outbound socket binding for the remote store into `socket-binding-group` element configuration:
++
+```xml
+<outbound-socket-binding name="remote-cache">
+    <remote-destination host="${remote.cache.host:localhost}" port="${remote.cache.port:11222}"/>
+</outbound-socket-binding>
+```
++
+
+.. The configuration of distributed cache `authenticationSessions` and other caches is left unchanged.
+
+.. Optionally enable DEBUG logging under the `logging` subsystem:
++
+```xml
+<logger category="org.keycloak.cluster.infinispan">
+    <level name="DEBUG"/>
+</logger>
+<logger category="org.keycloak.connections.infinispan">
+    <level name="DEBUG"/>
+</logger>
+<logger category="org.keycloak.models.cache.infinispan">
+    <level name="DEBUG"/>
+</logger>
+<logger category="org.keycloak.models.sessions.infinispan">
+    <level name="DEBUG"/>
+</logger>
+```
++
+
+. Copy the `NODE11` to 3 other directories referred later as `NODE12`, `NODE21` and `NODE22`.
+ 
+. Start `NODE11` :
++
+```
+cd NODE11/bin
+./standalone.sh -c standalone-ha.xml -Djboss.node.name=node11 -Djboss.site.name=site1 \
+-Djboss.default.multicast.address=234.56.78.1 -Dremote.cache.port=12232 -Djava.net.preferIPv4Stack=true \
+-Djboss.socket.binding.port-offset=3000
+
+```
++
+
+. Start `NODE12` :
++
+````
+cd NODE12/bin
+./standalone.sh -c standalone-ha.xml -Djboss.node.name=node12 -Djboss.site.name=site1 \
+-Djboss.default.multicast.address=234.56.78.1 -Dremote.cache.port=12232 -Djava.net.preferIPv4Stack=true \
+-Djboss.socket.binding.port-offset=4000
+````
++
+The cluster nodes should be connected. Something like this should be in the log of both NODE11 and NODE12:
++
+```
+Received new cluster view for channel keycloak: [node11|1] (2) [node11, node12]
+```
+NOTE: The channel name in the log might be different.
+
+. Start `NODE21` :
++
+```
+cd NODE21/bin
+./standalone.sh -c standalone-ha.xml -Djboss.node.name=node21 -Djboss.site.name=site2 \
+-Djboss.default.multicast.address=234.56.78.2 -Dremote.cache.port=13232 -Djava.net.preferIPv4Stack=true \
```

Comment:
[EX] Same as above - Replace `-Dremote.cache.port=13232` with `-Dremote.cache.host=jdg2` and remove the backslash and the following line